apps/emqx_gateway/src/emqx_gateway_api_clients.erl

```diff
@@ -234,8 +238,17 @@ subscriptions(delete, #{
     }
 }) ->
     with_gateway(Name0, fun(GwName, _) ->
-        _ = emqx_gateway_http:client_unsubscribe(GwName, ClientId, Topic),
-        {204}
+        case lookup_topic(GwName, ClientId, Topic) of
+            {ok, _} ->
+                case emqx_gateway_http:client_unsubscribe(GwName, ClientId, Topic) of
+                    {error, not_found} ->
+                        return_http_error(404, "Client not found");
+                    ok ->
+                        {204}
+                end;
+            {error, not_found} ->
```

Comment:
From the clauses, seems like `lookup_topic` might return other errors.  Maybe we need to add `{error, Reason}` here as well?

Reply:
The other ones I don’t want to handle, it’s a `500` situation then.